### ISSUE TITLE
[Feature Request] Support payments in other channels

### ISSUE THREAD

**Issue by Ma3yTa** [Microsoft/BotBuilder#4073]
Payments work only in WebChat and Skype  
https://docs.microsoft.com/en-us/bot-framework/dotnet/bot-builder-dotnet-request-payment  .
https://singularika.com/chatbots/payments_in_chatbots/
Payments with platform UI instead redirecting to url feels more native.
Please add support for Telegram, Facebook.
https://core.telegram.org/bots/payments
https://developers.facebook.com/docs/messenger-platform/payments

**Comment by Ma3yTa** [comment]
@willportnoy @Stevenic Is it possible to unify payments API from different channels?

**Comment by nwhitmont** [comment]
@Ma3yTa For reference, what is your development language/platform?

**Comment by Ma3yTa** [comment]
@eanders-MS @craigjensen just interesting are you guys already working on this functionality for BotBuilder v4?
As I saw It can be implemented for Facebook by sending a payload in ``channelData`` https://github.com/Microsoft/BotBuilder/issues/1859.
And for Telegram by calling ``sendInvoice`` with ``Payments provider token`` https://core.telegram.org/bots/api#sendinvoice

**Comment by Ma3yTa** [comment]
@eanders-MS @craigjensen If you didn't start implementing of this functionality yet - maybe I can help you to translate ``application/vnd.microsoft.card.payment`` to specific platform requests if you provide me access to connector's (Telegram, Facebook) source code.

**Comment by Ma3yTa** [comment]
@nwhitmont Could you please ask developers will they implement this because I don't know what to do - wait for implementation by BotBuilder team or implement by myself?

**Comment by Ma3yTa** [comment]
@MNHarun yes. it is possible. for example, you can use PayPal https://www.microsoft.com/developerblog/2016/10/31/integrating-payments-with-bots/ 
https://github.com/paypal/PayPal-NET-SDK
But Payments with Platform API looks more natural and not need to redirect user to web browser:
![](https://singularika.com/wp-content/uploads/2017/07/14130025_327619210923721_919324010_n.png)
![](https://singularika.com/wp-content/uploads/2017/07/pay3.jpg)
